### DISCUSSION TITLE
How to configure the default --network?

### DISCUSSION THREAD

**Discussion by rgl** [containerd/nerdctl#1128] [Q&A]
Is there a way to configure the default network used by `nerdctl`? I'm trying to use `nerdctl` with [`k0s`](https://github.com/k0sproject/k0s) configured `containerd`, where the default network is called `mynet` and always having to use `nerdctl --network=mynet` is a nuisance.

By reading the docs at https://github.com/containerd/nerdctl/blob/master/docs/config.md I did not find a way.

**Comment by Zheaoli** [comment]
emmmm, If I remember correctly, we don't have the way to explicitly spec an existed CNI network config as the default network. Maybe we can make this a feature? cc @junnplus

**Reply by junnplus** [reply]
> Maybe we can make this a feature? cc @junnplus

SGTM

**Comment by fahedouch** [comment]
We can append this list  https://github.com/containerd/nerdctl/blob/master/docs/config.md#properties by either adding a new field in `config.toml` file or CLI env var to set the default network

**Reply by Zheaoli** [reply]
SGTM, I'm working on it

**Comment by AkihiroSuda** [comment]
Implemented in
- https://github.com/containerd/nerdctl/pull/1092

**Reply by junnplus** [reply]
A little different, #1092 just persists the default network(bridge), the user can modify it, but @rgl may want to specify a different default network than bridge.

**Comment by rgl** [comment]
I would like to configure nerdctl to use an existing network (e.g. created by k0s/kube-router) without using the - -net cmd line parameter (but if it is specified, it would use it instead of the default network set in the config or environment variable (if that exists at all)).
